src/filesys.cpp

```diff
@@ -382,15 +382,22 @@ std::string TempPath()
 
 #endif
 
-void GetRecursiveSubPaths(const std::string &path, std::vector<std::string> &dst)
+void GetRecursiveSubPaths(const std::string &path, std::vector<std::string> &dst, bool list_files)
 {
 	std::vector<DirListNode> content = GetDirListing(path);
 	for(unsigned int  i=0; i<content.size(); i++){
 		const DirListNode &n = content[i];
 		std::string fullpath = path + DIR_DELIM + n.name;
-		dst.push_back(fullpath);
-		if (n.dir) {
-			GetRecursiveSubPaths(fullpath, dst);
+		if (list_files) {
+			dst.push_back(fullpath);
+			if (n.dir) {
+				GetRecursiveSubPaths(fullpath, dst, true);
+			}
+		else
+			if (n.dir) {
+				dst.push_back(fullpath);
+				GetRecursiveSubPaths(fullpath, dst, false);
+			}
```

Comment:
the way I did it was simpler. Also, even if you went this way, this code doesn't do what you expect. This is what your code does:

```cpp
if (list_files) {
	dst.push_back(fullpath);
	if (n.dir) {
		GetRecursiveSubPaths(fullpath, dst, true);
	} else if (n.dir) {
		dst.push_back(fullpath);
		GetRecursiveSubPaths(fullpath, dst, false);
	}
}
```